packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ChartAnnotationController.java

```diff
@@ -117,15 +130,39 @@ public ResponseEntity<ChartAnnotation> createChartAnnotation(
 			currentUserService.get().getId(),
 			projectId
 		);
+
+		final ChartAnnotation chartAnnotation;
 		try {
-			return ResponseEntity.status(HttpStatus.CREATED).body(
-				chartAnnotationService.createAsset(item, projectId, permission)
-			);
+			chartAnnotation = chartAnnotationService.createAsset(item, projectId, permission);
 		} catch (final IOException e) {
 			final String error = "Unable to create chart annotation";
 			log.error(error, e);
 			throw new ResponseStatusException(org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR, error);
 		}
+
+		final ClientEvent<ChartAnnotation> event = ClientEvent.<ChartAnnotation>builder()
+			.type(ClientEventType.CHART_ANNOTATION_CREATE)
+			.data(chartAnnotation)
+			.build();
+
+		try {
+			final RebacProject rebacProject = new RebacProject(projectId, reBACService);
```

Comment:
This looks like the same code as in the workflow sync messages?
If we're already reusing it we should move ahead with @YohannParis 's suggestion and move this into some sort of shared class.  Maybe not for this task, but, would you mind opening a new ticket to extract this and the code in `WorkflowController`?

Reply:
Yeah I think we should. I created a ticket for that: https://github.com/DARPA-ASKEM/terarium/issues/4806